docs/clean/user_guide/clean_stage_within_testscript.rst

```diff
@@ -0,0 +1,221 @@
+
+    (snip)
+
+    %AETEST-INFO: +......................................................................+
+    %AETEST-INFO: :    Starting STEP 3: Save the running config to the startup config    :
+    %AETEST-INFO: +......................................................................+
+
+    (snip)
+
+    %AETEST-INFO: +............................................................................+
+    %AETEST-INFO: :    Starting STEP 4: Verify next reload boot variables are correctly set    :
+    %AETEST-INFO: +............................................................................+
+
+    (snip)
+
+    %AETEST-INFO: +................................................................+
+    %AETEST-INFO: :    Starting STEP 5: Installing image 'bootflash:/image.bin'    :
+    %AETEST-INFO: +................................................................+
+
+    (snip)
+
+    %AETEST-ERROR: Failed reason: Failed to install the image.
+    %AETEST-INFO: The result of STEP 5: Installing image 'bootflash:/image.bin' is => FAILED
+    %AETEST-INFO: The result of section my_section is => PASSED
+    %AETEST-INFO: The result of testcase Test is => PASSED
+
+    %EASYPY-INFO: +----------------------------------------------------------------------------+
+    %EASYPY-INFO: |                            Task Result Summary                             |
+    %EASYPY-INFO: +----------------------------------------------------------------------------+
+    %EASYPY-INFO: Task-1: testscript.Test                                                 PASSED
+    %EASYPY-INFO:
+    %EASYPY-INFO: +----------------------------------------------------------------------------+
+    %EASYPY-INFO: |                            Task Result Details                             |
+    %EASYPY-INFO: +----------------------------------------------------------------------------+
+    %EASYPY-INFO: Task-1: testscript
+    %EASYPY-INFO: `-- Test                                                                PASSED
+    %EASYPY-INFO:     `-- my_section                                                      PASSED
+
+.. note::
+
+    There are two **default but extremely important** behaviours to note:
+
+        * Results from the stage do not roll-up to affect the testscript
+        * The steps of the stage will not appear in the `Task Result Details`
+
+    Both of these behaviours can be observed in the above snippet.
+
+To enable this roll-up, pass the steps object which is automatically created by the pyATS infrastructure as shown below.
+
+.. code-block:: python
+    :linenos:
+    :emphasize-lines: 6,12
+
+    from pyats import aetest
+
+    class MyTestcase(aetest.Testcase):
+
+        @aetest.test
+        def my_test(self, steps, testbed):
+
+            device = testbed.devices['uut']
+            device.connect()
+
+            device.api.clean.install_image(
+                steps=steps
```

Comment:
nested steps also works? might not need the example, but might better to have one sentence about it if supported..